solutionbox/code_free_ml/transform_raw_data.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Flake8 cannot disable a warning for the file. Flake8 does not like beam code
+# and reports many 'W503 line break before binary operator' errors. So turn off
+# flake8 for this file.
+# flake8: noqa
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import datetime
+import json
+import logging
+import os
+import random
+import sys
+import apache_beam as beam
+from apache_beam.metrics import Metrics
+import six
+from tensorflow.python.lib.io import file_io
+from tensorflow_transform import coders
+from tensorflow_transform.beam import impl as tft
+from tensorflow_transform.beam import tft_beam_io
+from tensorflow_transform.tf_metadata import metadata_io
+
```

Comment:
Add some quick comments about why transform?

1. Transform is required if a) there are img_url_to_vec transformation. b) data is in BigQuery.
2. Transform is desired if data transformation is expensive, such as tf-idf is used on text.